kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/TestCoroutineInterceptor.kt

```diff
@@ -28,7 +29,15 @@ class TestCoroutineInterceptor : TestExecutionInterceptor {
    ): TestResult {
       var result: TestResult = TestResult.Ignored()
       logger.log { Pair(testCase.name.testName, "Switching context to coroutines runTest") }
-      runTest {
+
+      // Handle timeouts here to avoid the influence of the default timeout set inside runTest
+      val timeout = if (testCase.type == TestType.Container) {
```

Comment:
should we delegate this logic to `testCase` so that we avoid its tight coupling with this interseptor?